packages/code-du-travail-api/src/server/routes/search/index.js

```diff
@@ -54,24 +48,49 @@ router.get("/search", async ctx => {
   const size = Math.min(ctx.request.query.size || MAX_RESULTS, 100);
   const body = getSearchBody({ query, size: size, excludeSources });
   const facetBody = getFacetsBody({ query });
-  const [esResults, semResults] = await Promise.all([
-    elasticsearchClient.search({ index, body }),
+
+  const esResponse = new Promise((resolve, reject) => {
+    const timer = setTimeout(() => {
+      reject([]);
+    }, MAX_TIMEOUT);
+    elasticsearchClient
+      .search({ index, body })
+      .then(val => {
+        clearTimeout(timer);
+        resolve(val);
+      })
+      .catch(err => {
+        reject(err);
+        clearTimeout(timer);
+      });
+  });
+
+  const semResponse = new Promise((resolve, reject) => {
+    const timer = setTimeout(() => {
+      reject([]);
+    }, MAX_TIMEOUT);
     fetch(
-      `${process.env.NLP_URL}/api/search?q=${query}&excludeSources=${excludeSources}`,
-      { signal: controller.signal }
+      `${process.env.NLP_URL}/api/search?q=${query}&excludeSources=${excludeSources}`
     )
-      .then(data => data.json())
-      .catch(err => {
-        if (err.name === "AbortError") {
-          console.log("error was aborted after timeout");
-          return [];
-        }
-      })
-      .finally(() => {
-        clearTimeout(timeout);
+      .then(val => {
+        clearTimeout(timer);
+        resolve(val);
       })
-  ]);
+      .catch(err => {
+        reject(err);
+        clearTimeout(timer);
+      });
+  });
+  // utils which returns a status whatever
+  const reflect = p =>
+    p.then(v => ({ v, timedout: false }), e => ({ e, timedout: true }));
 
+  const [esResults, semResults] = await Promise.all([
```

Comment:
Stylé, ça fonctionne alors ??

Reply:
c'est le turfu

Reply:
T'as vu un peu le trick qu'il a utilisé pour wrap ses promesses dans un `reflect` pour s'assurer que le promise all ne fail jamais ??? Armand is on F-I-R-E

Reply:
you can use default value here 
```
also catching and return empty [] before then should work as you intend and avoid  reflect() function